### PR TITLE
eslint fixes

### LINKED ISSUE #144: `npm run lint` fails
Haven't checked into it; we should make it work again.

### PR DESCRIPTION
**Pull request by threepointone**
fixes these lint errors, fixes #144 
```
demos/demo6/babel.js
  0:0  error  Maximum call stack size exceeded

native/index.js
   1:0   error  Strict mode is not permitted                                     strict
   7:0   error  Unexpected var, use let or const instead                         no-var
   7:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
   9:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
   9:0   error  Unexpected var, use let or const instead                         no-var
  11:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  11:0   error  Unexpected var, use let or const instead                         no-var
  13:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  13:0   error  Unexpected var, use let or const instead                         no-var
  15:0   error  Unexpected var, use let or const instead                         no-var
  15:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  17:0   error  Unexpected var, use let or const instead                         no-var
  17:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  19:0   error  Unexpected var, use let or const instead                         no-var
  19:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  19:18  error  ["default"] is better written in dot notation                    dot-notation
  19:45  error  ["default"] is better written in dot notation                    dot-notation
  21:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  21:0   error  Unexpected var, use let or const instead                         no-var
  22:0   error  Unexpected var, use let or const instead                         no-var
  22:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  26:0   error  Unexpected var, use let or const instead                         no-var
  26:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  28:0   error  Unexpected var, use let or const instead                         no-var
  28:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  30:18  error  ["default"] is better written in dot notation                    dot-notation
  31:0   error  Unexpected var, use let or const instead                         no-var
  31:0   error  All "var" declarations must be at the top of the function scope  vars-on-top
  32:15  error  ["default"] is better written in dot notation                    dot-notation
  32:42  error  Missing trailing comma                                           comma-dangle

test/Spring-test.js
  354:8  error  Do not use setState in componentDidMount  react/no-did-mount-set-state

test/createMockRaf.js
  30:1  error  Unnecessary semicolon  no-extra-semi

✖ 33 problems (33 errors, 0 warnings)
```

### CODE REVIEW

File: test/Spring-test.js [hunk 1]
```diff
@@ -351,6 +351,9 @@ describe('Spring', () => {
       },
 
       componentDidMount() {
+        this.update();
```
Comment (chenglou) [inline test/Spring-test.js]: We know what we're doing. Can we disable this rule that warns this please?

Comment (chenglou) [inline test/Spring-test.js]: Screw this, I'll do it, thanks! =)

Comment (chenglou) [inline test/Spring-test.js]: Actually no, you do it, lol

Comment (threepointone) [inline test/Spring-test.js]: the rule is `Do not use setState in componentDidMount  react/no-did-mount-set-state` are you sure?

Comment (chenglou) [inline test/Spring-test.js]: On second/third thought, let's keep it this way. I'm pretty sure I'd be responsible with it if I ever need it, but let's dissuade potential contributors from using this pattern.

**Pull request merged**